### DISCUSSION TITLE
Writing solutions with pySCIPopt

### DISCUSSION THREAD

**Discussion by abfariah** [ambros-gleixner/MIPcc23#22] [Ideas]
Hi everyone, 

To other users of the pySCIPopt package : the solution provided in the examples/python_example/run.py file to write the best primal solution in a __.sol file is not suited for series with a large number of variables. For rhs obj series 1(>90e3 variables), it takes 15 to 25 min to write the problem solution. 
Using the writeBestSol(filename : str, write_zeros : bool) method from pyscipopt.Model, this time is reduced to 0.013s. 

To the organizers : is it ok to change the approach for writing down solutions? The .sol format is very close to the one you describe in RULES.md. 

Here is a sample on rhs_obj_1 with write_zeros = False: 

![image](https://user-images.githubusercontent.com/90516964/224001172-87f2c8cf-e1fb-4d81-b1e3-154a275cc515.png)

To be compared to the examples/python_example/run.py version:

![image](https://user-images.githubusercontent.com/90516964/224000995-e4a35668-237a-4060-a0e0-d8b1f169d6b1.png)

**Comment by ambros-gleixner** [comment]
Hi Paul, thank you very much for noting this! I just pushed an update to the Python example that should also print solutions much faster. Can you check? Is there still a huge difference to the `writeBestSol()` method?

**Reply by abfariah** [reply]
Hi Ambros, the writing time on rhs obj 1 is now 0.31s, which is longer but totally fine. My main worry was to exceed the strict 1.1 * t global time limit of the competition, this is not a concern anymore. 

Thank your for such a swift response !

**Reply by ambros-gleixner** [reply]
Okay, happy to hear. Using `writeBestSol()` would also have been okay, but then I would have suggested a quick `sed` command to fix the formatting.

**Reply by abfariah** [reply]
A simple detail, I compared the .sol files produced by both methods and I noticed that with the "manual" solution, there were entries in the solution files with 1e-14 values. These entries were not present in the file produced by SCIP's `writeBestSol()` methods, meaning they were interpreted as 0.

To rectify this, it may be best to write for example :

`if abs(solval) > 1e-10:`

instead of :

`if solval != 0.0:`

in the loop on variables.

**Reply by ambros-gleixner** [reply]
True, thanks!
